File: src/main/scala/com/abaddon83/cardsGames/shares/games/Game.scala
package com.abaddon83.cardsGames.shares.games

import com.abaddon83.cardsGames.shares.players.Player

trait Game {
  protected val gameIdentity: GameIdentity
  protected val maxPlayers: Int
  protected val minPlayers: Int
  protected val totalCardsRequired: Int
  protected val players: List[Player]

  def numPlayers: Int = {
    players.size
  }

  def listOfPlayers(): List[Player]= {
    players
  }

  def identity(): GameIdentity ={
    gameIdentity
  }
}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/models/games/Game.kt
package com.abaddon83.burracoGame.commandModel.models.games

import com.abaddon83.burracoGame.commandModel.models.players.Player

interface Game{
    val maxPlayers: Int
    val minPlayers: Int
    val totalCardsRequired: Int
    val players: List<Player>
    fun numPlayers(): Int = players.size
    fun listOfPlayers(): List<Player> = players

}
